crates/miden-tx/src/host/mod.rs

```diff
@@ -652,7 +672,7 @@ where
                     current_account_id,
                     // Slot index should always fit into a usize.
                     slot_index: slot_index.as_int() as usize,
-                    map_root,
+                    current_map_root,
                     map_key,
```

Comment:
Related to the previous comments: maybe it could be somewhere around here that we check if we are dealing with the native account or not and act accordingly.